test/bridge/testase.py

```diff
@@ -0,0 +1,236 @@
+        assert np.allclose(
+            ase_data.atomcharges["mulliken"], data.atomcharges["mulliken"]
+        )
+        assert np.allclose(ase_data.atomspins["mulliken"], 0)
+        assert np.allclose(ase_data.atommasses, data.atommasses)
+        assert np.isclose(ase_data.charge, data.charge, atol=1e-5)
+        assert np.isclose(ase_data.mult, data.mult)
+        assert np.isclose(ase_data.natom, len(data.atomnos))
+        assert np.isclose(ase_data.temperature, 0)
+
+    def test_write_and_read_trivial_trajectories(self):
+        """Ensure write and read trajectory files with single structures."""
+        # an open-shell single point calculation
+        data = ccopen("data/ORCA/basicORCA4.2/dvb_sp_un.out").parse()
+        cclib2ase.write_trajectory("dvb_sp_un.traj", data)
+        trajdata = cclib2ase.read_trajectory("dvb_sp_un.traj")
+
+        assert np.allclose(trajdata.atomcoords, data.atomcoords)
+        assert np.allclose(trajdata.scfenergies, data.scfenergies)
+        # no grads here
+
+        assert np.allclose(trajdata.atomnos, data.atomnos)
+        assert np.allclose(trajdata.atommasses, data.atommasses)
+        assert np.allclose(trajdata.natom, data.natom)
+        assert np.allclose(trajdata.charge, data.charge)
+        assert np.allclose(trajdata.mult, data.mult)
+        assert np.allclose(trajdata.moments, data.moments)
+
+        # no temperature here
+        # no freeenergy here
+
+        assert np.allclose(
+            trajdata.atomcharges["mulliken"], data.atomcharges["mulliken"]
+        )
+        assert np.allclose(
+            trajdata.atomspins["mulliken"], data.atomspins["mulliken"]
+        )
+
+        # a closed-shell single structure frequency calculation
+        data = ccopen("data/ORCA/basicORCA4.2/dvb_ir.out").parse()
+        cclib2ase.write_trajectory("dvb_ir.traj", data)
+        trajdata = cclib2ase.read_trajectory("dvb_ir.traj")
+
+        assert np.allclose(trajdata.atomcoords, data.atomcoords)
+        assert np.allclose(trajdata.scfenergies, data.scfenergies)
+        # no grads here
+
+        assert np.allclose(trajdata.atomnos, data.atomnos)
+        assert np.allclose(trajdata.atommasses, data.atommasses)
+        assert np.allclose(trajdata.natom, data.natom)
+        assert np.allclose(trajdata.charge, data.charge, atol=1e-5)
+        assert np.allclose(trajdata.mult, data.mult)
+        assert np.allclose(trajdata.moments, data.moments)
+
+        # no temperature here
+        # no freeenergy here
+
+        assert np.allclose(
+            trajdata.atomcharges["mulliken"], data.atomcharges["mulliken"]
+        )
+        # no atomspins here
+
+    def test_write_and_read_opt_trajectories(self):
+        """Ensure write and read trajectory files with optimizations."""
+        # geometry optimization
+        data = ccopen("data/ORCA/basicORCA4.2/dvb_gopt.out").parse()
+        cclib2ase.write_trajectory("dvb_gopt.traj", data)
```

Comment:
Do you have any native ASE trajectory you could use for testing here?

Reply:
I generated a simple optimization trajectory and included in the tests. The trajectory file was generated as follows:

```python3
from ase import Atoms
from ase.optimize import BFGS
from ase.calculators.orca import ORCA

h2o = Atoms("OH2", positions=[[0, 0, 0], [1, 0, 0], [0, 1, 0]])
h2o.calc = ORCA(orcasimpleinput="PM3")
opt = BFGS(h2o, trajectory="h2o.traj")
opt.run(fmax=0.02)
# =>
#       Step     Time          Energy         fmax
# BFGS:    0 16:00:28     -324.613009        4.3722
# BFGS:    1 16:00:28     -324.787779        1.4193
# BFGS:    2 16:00:28     -324.853574        1.0747
# BFGS:    3 16:00:29     -324.903836        0.4422
# BFGS:    4 16:00:29     -324.907186        0.1344
# BFGS:    5 16:00:29     -324.907382        0.0226
# BFGS:    6 16:00:30     -324.907387        0.0000
```